### PR TITLE
Pls work tests

### PR DESCRIPTION
**Pull request by aesthetiques**

### CODE REVIEW

File: caleb-lab/test/test.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          });
+          // client.end();
+          done();
+        });
+      });
+    });
+  });
+});
+
+// expect(client.socket.write(`@all this should print things.`).to.be.equal.to(`this should print things.`));
```
Comment (ghost) [inline caleb-lab/test/test.js]: DUde! This is effing sick!

File: caleb-lab/package.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "caleb-lab",
+  "version": "1.0.0",
+  "description": "schizophrenic tcp chat[this only works if you talk to yourself in node]",
+  "main": "index.js",
+  "scripts": {
+    "test": "mocha",
```
Comment (JudyVue) [inline caleb-lab/package.json:7]: **TODO FOR LOST POINTS**
You need a ``start`` script here, i.e.:

```
"start": "nodemon server.js"
```

File: caleb-lab/server.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    if(c.nickName === string.split(' ').shift().trim()){
+      //print your message on your screen
+      client.socket.write(`to ${c.nickName}: ${string.split(' ').slice(1).join(' ')}`);
+      //print your message on their screen
+      c.socket.write(`from ${client.nickName}: ${string.split(' ').slice(1).join(' ')}`);
+    }
+  });
+});
+//to get the commands
+ee.on('@help', (client) => {
```
Comment (JudyVue) [inline caleb-lab/server.js]: This ``@help`` command is awesome and great for UX! 👍

File: caleb-lab/server.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  client.socket.write(`Commands:\nGlobal Channel: @all (message)\n Private message: @pm (username)\nChange nickname: @nick (name)\n`);
+});
+
+//check the net module for the premade EventEmitters that exist on server
+server.on('connection', socket => {
+  let client = new Client(socket);
+  pool.push(client);
+  pool.forEach(c => c.socket.write(`${client.userName} has connected! Samuel says: There are too many motha fuckin' clients on this motha fuckin' server!!11!\n`));
```
Comment (JudyVue) [inline caleb-lab/server.js]: `` Samuel says: There are too many motha fuckin' clients on this motha fuckin' server`` 👍

File: caleb-lab/server.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    ee.emit('default', client, data.toString());
+  });
+
+  socket.on('close', function(){
+    console.log(`${client.nickName} has left the room.\n`);
+    pool.forEach(client.socket.write(`${client.nickName} has left the room.\n`));
+    removeSocket(socket);
+  });
+});
```
Comment (JudyVue) [inline caleb-lab/server.js]: **TODO FOR LOST POINTS**

Lines 70-75 are breaking your program when a user exits. Change to this:

```
  socket.on('close', function(){
    console.log(`${client.nickName} has left the room.\n`);
    // pool.forEach(client.socket.write(`${client.nickName} has left the room.\n`));
    removeSocket(socket);
    pool.forEach(c => `${c.nickName} has left the room.\n` );
    console.log('this is the new pool', pool);
  });
```

Lines 72 isn't written quite correctly so it breaks your code. Your ``removeSocket`` helper function works correctly, so great job splicing from the client pool array.

File: caleb-lab/test/test.js (hunk 1 above)
Comment (JudyVue) [inline caleb-lab/test/test.js]: **TODO FOR LOST POINTS**

Take out commented code and console.logs.

File: caleb-lab/test/test.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
+'use strict';
```
Comment (JudyVue) [inline caleb-lab/test/test.js:1]: This test module looks great. 👍

File: caleb-lab/.gitignore [hunk 7]
```diff
@@ -0,0 +1,58 @@
+# Logs
```
Comment (JudyVue) [inline caleb-lab/.gitignore:22]: **TODO FOR LOST POINTS**

Top-level comment to let you know you need a ``README.md`` file, aka docs, per the instructions:

```
Write docs on how to get the project running
Write docs on how to connect to the server
```

**Pull request closed without merging**